org.opentosca.planbuilder.postphase.plugin.instancedata/src/org/opentosca/planbuilder/postphase/plugin/instancedata/bpel/Handler.java

```diff
@@ -330,6 +340,265 @@ public boolean handleTerminate(final BPELPlanContext context, final AbstractNode
+
+            if (preState != null) {
+
+                try {
+
+                    // assign prestate to state variable
+                    final BPELProcessFragments frag = new BPELProcessFragments();
+                    Node assignNode = frag.createAssignXpathQueryToStringVarFragmentAsNode("assignRelationStateFor_"
+                        + operationName + "_" + System.currentTimeMillis(), "string('" + preState + "')", stateVarName);
+                    assignNode = context.importNode(assignNode);
+                    lastSetState = preState;
+
+                    // assign the state before the assign of the invoker request
+                    // is made
+                    final Node bpelAssignNode =
+                        assignContentElement.getParentNode().getParentNode().getParentNode().getParentNode();
+                    bpelAssignNode.getParentNode().insertBefore(assignNode, bpelAssignNode);
+
+                    // create REST Put activity
+                    final String bpelString =
+                        this.fragments.generateBPEL4RESTLightPUTInstanceState(relationInstanceURLVarName, stateVarName);
+                    Node extActiv = ModelUtils.string2dom(bpelString);
+                    extActiv = context.importNode(extActiv);
+
+                    // send the state before the assign of the invoker request
+                    // is made
+                    bpelAssignNode.getParentNode().insertBefore(extActiv, bpelAssignNode);
+                }
+                catch (final IOException e2) {
+                    e2.printStackTrace();
+                }
+                catch (final SAXException e) {
+                    e.printStackTrace();
+                }
+                catch (final ParserConfigurationException e) {
+                    e.printStackTrace();
+                }
+            }
+
+            if (postState != null) {
+                try {
+                    // create state assign activity
+                    final BPELProcessFragments frag = new BPELProcessFragments();
+                    Node assignNode =
+                        frag.createAssignXpathQueryToStringVarFragmentAsNode("assignRelationState_" + operationName + "_"
+                            + System.currentTimeMillis(), "string('" + postState + "')", stateVarName);
+                    assignNode = context.importNode(assignNode);
+
+                    lastSetState = postState;
+
+                    /*
+                     * assign the state after the receiving the response of the
+                     */
+
+                    // fetch assign node
+                    final Node bpelAssignNode =
+                        assignContentElement.getParentNode().getParentNode().getParentNode().getParentNode();
+
+                    // fetch the variable name which is used as request body
+                    final String reqVarName = fetchRequestVarNameFromInvokerAssign(assignContentElement);
```

Comment:
comment becomes unnecessary when changing variable name to `requestBodyVariableName`